### ISSUE TITLE
The Number of Shifted Bits

### ISSUE THREAD

**Issue by mengjingyouling** [mostafaelhoushi/DeepShift#17]
The weight value is cliped to [- 1, 1] in the code：
**self.shift_range = (-1 * (2**(weight_bits - 1) - 2), 0)**

But the weight should have a small number of values exceeding 1.
For example 2 , it  is easily obtained by shifting,<<1.
Is it better to rewrite the code as follows：
**self.shift_range = (-1 * (2**(weight_bits - 1) - 2), 2**(weight_bits)**

Thank you. @mostafaelhoushi

**Comment by mengjingyouling** [comment]
For hardware, the power consumption of left and right shift is the same, but the left and right shift can represent a wider range.

**Comment by mostafaelhoushi** [comment]
Thanks @mengjingyouling for your idea.
The reason I did that is that I found in regular neural networks weights are most of the time a fractional value between -1 and 1. 

I agree it might be worth it to try values slightly bigger than 1 or 2, but not bigger than that, as weights will rarely reach such large values.

Also, note that we want to constrain the possible shift values to the number of weight buts available to us. Hence your proposal can be something like:
```
self.shift_range = (-1 * (2**(weight_bits - 2) -1), 2**(weight_bits - 2)
```
This needs to be refined it to ensure that the number of possible values won't exceed 2**weights.
